compiler-plugin/src/main/kotlin/arrow/meta/phases/analysis/DefaultElementScope.kt

```diff
@@ -55,6 +60,9 @@ class DefaultElementScope(project: Project) : ElementScope {
   override val String.typeOrNull: Scope<KtTypeReference>
     get() = Scope(delegate.createTypeIfPossible(trimMargin()))
 
+  override val String.nullableType: NullableTypeScope
+    get() = TODO("not implemented") //To change initializer of created properties use File | Settings | File Templates.
```

Comment:
@raulraja @bloderxd There appears to be no 1:1 match in the PsiElementFactory (the delegate) that creates a `KtNullableType`. There are delegate methods that create very close types (see the ones surrounding this bit of code - I wonder if it makes sense to remove this from ElementScope then.

I briefly looked around for uses in the kotlin codebase and I think it's maybe like not a "top-level" element but it may be good to still have this:

<img width="1220" alt="Screen Shot 2019-11-13 at 11 47 54 AM" src="https://user-images.githubusercontent.com/6917498/68789664-8391d180-060b-11ea-8dea-ca56adc22c03.png">

Please, let me know what ya'll think!

Reply:
As far as  I understand a KtNullableType is an instance of a KtTypeElement which is extra info on a KtTypeReference. What you render as a type is the KtTypeReference so I'm not sure we need a template for KtNulllableType but we need one for KtTypeReference and KtElementType which is the reference.

Reply:
We can close this one then!